lib/saluki-components/src/destinations/datadog_metrics/endpoint/endpoints.rs

```diff
@@ -113,16 +116,42 @@ impl ResolvedEndpoint {
         Ok(Self {
             endpoint,
             api_key: api_key.to_string(),
+            config: None,
         })
     }
 
+    /// Creates a new  `ResolvedEndpoint` instance from an existing `ResolvedEndpoint`, adding an optional `RefreshableConfiguration`.
+    pub fn with_refreshable_configuration(self, config: Option<RefreshableConfiguration>) -> Self {
+        Self {
+            endpoint: self.endpoint,
+            api_key: self.api_key,
+            config,
+        }
+    }
+
     /// Returns the endpoint of the resolver.
     pub fn endpoint(&self) -> &Url {
         &self.endpoint
     }
 
     /// Returns the API key associated with the endpoint.
```

Comment:
```suggestion
    /// Returns the API key associated with the endpoint.
    ///
    /// If a refreshable configuration has been configured, the API key will be queried from the
    /// configuration and stored if it has been updated since the last time `api_key` was called.
```